modules/cluster/irsa.tf

```diff
@@ -378,3 +378,64 @@ module "iam_assumable_role_bucketrepo" {
   role_policy_arns              = [var.create_bucketrepo_role ? aws_iam_policy.bucketrepo[0].arn : ""]
   oidc_fully_qualified_subjects = ["system:serviceaccount:${local.jenkins-x-namespace}:bucketrepo-bucketrepo"]
 }
+
+// ----------------------------------------------------------------------------
+// External Secrets
+// ----------------------------------------------------------------------------
+data "aws_iam_policy_document" "secrets-policy" {
+  count = var.create_secrets_role ? 1 : 0
+  statement {
+    effect = "Allow"
+    actions = [
+      "secretsmanager:CreateSecret",
+      "secretsmanager:DescribeSecret",
+      "secretsmanager:GetResourcePolicy",
+      "secretsmanager:GetSecretValue",
+      "secretsmanager:ListSecretVersionIds",
+      "secretsmanager:ListSecrets",
+      "secretsmanager:PutSecretValue",
+      "secretsmanager:UpdateSecret",
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:secretsmanager:${var.region}:${local.project}:secret:secret/data/lighthouse/*",
+      "arn:${data.aws_partition.current.partition}:secretsmanager:${var.region}:${local.project}:secret:secret/data/jx/*",
+      "arn:${data.aws_partition.current.partition}:secretsmanager:${var.region}:${local.project}:secret:secret/data/nexus/*"
+    ]
+  }
+
+  statement {
```

Comment:
lets separate it into 2 different policies, one for asm and one for parameter store.

Reply:
Would I create two roles too? or one Role with both Policies attached?

Reply:
Two separate roles would be better imo.